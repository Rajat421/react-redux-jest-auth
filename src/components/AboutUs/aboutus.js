/**
 * Created by Counter on 6/29/2017.
 */
import React ,{Component} from 'react'
import Profile from './profile/profile'
import Team from './team/team'
import Contact from './contactus/contactus'
import Header from '../Header/header'
import {connect} from 'react-redux'
import * as actions from '../../actions/authActions'

export class AboutUs extends Component {
    constructor(props){
        super(props);
        if(!localStorage.getItem('key')){
            this.props.history.push('/')
        }

    }
    componentWillReceiveProps(newProps){
        console.log(newProps,'ppppp');
        if(!newProps.isAuthenticated){
            this.props.history.push('/')
        }
    }
    render(){

        return(
            <div>
                <Header logout={this.props.logout}/>
                {(this.props.location.pathname==="/aboutus")?<div>
                < h2 style={{textAlign:'center'}}><u>About Us</u></h2>
                    <div>
                    <Profile/>
                        </div>
                    <div>
                    <Team/>
                        </div>
                    <div>
                    <Contact/>
                        </div>


                    </div>:null
                }
            </div>
        )
    }
}

const mapStateToProps=(state)=>{


    return {
        isAuthenticated:state.auth.isAuthenticated
    }
}


export default connect(mapStateToProps,actions)(AboutUs);
