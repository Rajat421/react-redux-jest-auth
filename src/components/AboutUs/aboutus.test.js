
import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import {AboutUs} from "./aboutus";
import localStorageMock from "../../mock-local-storage";
import {MemoryRouter} from "react-router";

window.localStorage = localStorageMock;

test('AboutUs component should render as expected',()=>{
    const component =shallow(<MemoryRouter  initialEntries={ ['/aboutus'] }
                                            initialIndex={0}><AboutUs/></MemoryRouter>);
    const tree =toJson(component);
    expect(tree).toMatchSnapshot()
});
