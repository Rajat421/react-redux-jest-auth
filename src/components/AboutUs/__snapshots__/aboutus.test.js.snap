// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs component should render as expected 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "k32v4h",
          "pathname": "/aboutus",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "k32v4h",
        "pathname": "/aboutus",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <AboutUs />
</Router>
`;
